presto-hive/src/main/java/io/prestosql/plugin/hive/HivePageSource.java

```diff
@@ -406,6 +418,13 @@ public ConnectorPageSource getPageSource()
         throw new PrestoException(NOT_SUPPORTED, format("Unsupported coercion from %s to %s", fromHiveType, toHiveType));
     }
 
+    public static boolean narrowerThan(VarcharType first, VarcharType second)
+    {
+        requireNonNull(first, "first is null");
+        requireNonNull(second, "second is null");
+        return first.getLength().orElse(VarcharType.UNBOUNDED_LENGTH) < second.getLength().orElse(VarcharType.UNBOUNDED_LENGTH);
```

Comment:
UNBOUNDED_LENGTH is not really a length limit (otherwise, why would we discern this as a special case?)

would
```
if (first.isUnbounded() || second.isUnbounded()) {
    return !first.isUnbounded();
}
return first.getBoundedLength() < second.getBoundedLength();
```

work?